cmd/setupcommittee.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/0xPolygon/cdk-validium-node/config"
+	"github.com/0xPolygon/cdk-validium-node/etherman/smartcontracts/cdkdatacommittee"
+	"github.com/0xPolygon/cdk-validium-node/log"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+)
+
+func setupCommittee(ctx *cli.Context) error {
+	c, err := config.Load(ctx, true)
+	if err != nil {
+		return err
+	}
+	if !ctx.Bool(config.FlagYes) {
+		fmt.Println("*WARNING* Are you sure you want to setup committee? [y/N]: ")
+		var input string
+		if _, err := fmt.Scanln(&input); err != nil {
+			return err
+		}
+		input = strings.ToLower(input)
+		if !(input == "y" || input == "yes") {
+			return nil
+		}
+	}
+
+	setupLog(c.Log)
+
+	// Check if it is already registered
+	etherman, err := newEtherman(*c)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	// load auth from keystore file
+	addrKeyStorePath := ctx.String(config.FlagKeyStorePath)
+	addrPassword := ctx.String(config.FlagPassword)
+	authL1, _, err := etherman.LoadAuthFromKeyStore(addrKeyStorePath, addrPassword)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	dataCommitteeContract := c.DataAvailability.L1.DataCommitteeAddress
+	fmt.Println("cdkDataCommitteeContract: ", dataCommitteeContract)
+
+	clientL1, err := ethclient.Dial(c.DataAvailability.L1.RpcURL)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	dacSC, err := cdkdatacommittee.NewCdkdatacommittee(
+		common.HexToAddress(dataCommitteeContract),
+		clientL1,
+	)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	const nSignatures = 1
+	addrsBytes := []byte{}
+	urls := []string{}
+
+	// load dac member from keystore file
```

Comment:
I don't think we need to load the DAC member from the keystore file. also given that if we were to support multiple DAC member, we need to load multiple files and that will need multiple passwords to decrypt those files which can be too complicated.

can we keep it simple by just asking the user to specify the list of DAC member addresses and URLs in the config file?

Reply:
yes, in general, the admin is the one who sends the L1 tx to setup the committee, but this admin:

- shouldn't be part of the committee
- shouldn't know the private keys of the committee

this is intentional, as the hole idea is that the DAC and the admin of the network are different entities